src/net/net.c

```diff
@@ -220,3 +220,38 @@ int net_default_gateway_get(int af, struct sa *gw)
 
 	return 0;
 }
+
+
+/**
+ * Resolve a given address and returns an sa object
+ *
+ * @param addr  IP address (IPv4, IPv6 and IPv6 link local including the
+ *                          interface suffix)
+ * @param sa    Returned socket address
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int net_resolve_address(const char *addr, struct sa *sa)
+{
+	struct addrinfo *res, *res0;
+	int err;
+
+	err = getaddrinfo(addr, NULL, NULL, &res0);
```

Comment:
setting ai_flags AI_NUMERICHOST will disable DNS query:

```
                   AI_NUMERICHOST  If the AI_NUMERICHOST bit is set, it indicates that
                                    hostname should be treated as a numeric string
                                    defining an IPv4 or IPv6 address and no name reso-
                                    lution should be attempted.
```

see udp.c and tcp.c

Reply:
perfect, thanks!

Reply:
It works. Will add a test case to retest that also checks that resolution is disabled.

Should we always call `net_resolve_address()` from `sa_set_str()`?

Reply:
I tried this. The `net_inet_pton()` seems to convert IPv6-mapped IPv4 addresses to the associated IPv4 address. And `net_resolve_address()` does not. Is this conversion mandatory?

Reply:
Another problem. This test fails:
```
	err = net_resolve_address("6002", &sa);
	TEST_EQUALS(EADDRNOTAVAIL, err);
```
"6002" is not a valid IP address.

Should we add a `re_regex` check for IPv6 and one for IPv4 in `net_resolve_address()`?